### ISSUE TITLE
Raise error events?

### ISSUE THREAD

**Issue by jbergknoff** [tomas/needle#178]
I noticed that Needle streams don't emit `error` events. Instead, they will emit an `end` event with an error argument [1]. This makes it easy to miss/ignore stream errors, whereas an `error` event will crash the process if not handled. Is this intentional?

[1] https://github.com/tomas/needle/blob/2ce689a4313af25bbe7eb52d659b3d6c70454aff/lib/needle.js#L384

**Comment by tomas** [comment]
Yes and no. :)

What type of error would we miss or ignore? Wouldn't they still be passed on to the `end` event?

**Comment by tomas** [comment]
By the way, if you're referring to errors triggered by decompressors, we just got that merged into master via #172.

**Comment by jbergknoff** [comment]
Thank you for the response. No, I wasn't thinking of errors relating to decompressors.

I recently noticed that `ECONNRESET` ("socket hang up") errors show up in the `end` event. I had a listener on the stream's `error` event but not on `end`, and my application was hanging until I looked at the Needle source carefully.

Normally, this would be raised as an `error` event and then reach my listener (or crash the process if there is no listener -- the `error` event is [treated specially](https://github.com/nodejs/node/blob/1a1ff77feb3ed003e71bcbd066deadcaf9a82652/lib/events.js#L149)). Somewhat related: I've never seen parameters supplied on an `end` event before.
